lib/facility.rb

```diff
@@ -1,14 +1,86 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :collected_fees, :registered_vehicles
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    if facility_details.keys.include?(:dmv_office)
```

Comment:
consider extracting these chunks of data-cleaning logic into helper methods that can be called from the initialize. We generally want to keep methods pretty short (8-10) lines so when methods move beyond that, using helper methods can be a great resource! For example, your initialize could be cleaned up like this:

```
def initialize(facility_details)
    @name = populate_name(facility_details)
    @address = populate_address(facility_details)
    ...
end

def populate_name(data)
    if data.keys.include?(:dmv_office)
         return facility_details[:dmv_office]
    etc...
end

    
```

Reply:
That makes sense!